admin/data/sql/migrations/20190210134733.do.create-view-schools-average.sql

```diff
@@ -0,0 +1,31 @@
+CREATE VIEW [mtc_admin].[vewSchoolsAverage] AS
+
+SELECT
+  p.school_id,
+  CAST(SUM(
+      CASE
+      WHEN (pa.id IS NOT NULL) THEN 0
+      WHEN (latestPupilCheck.mark IS NOT NULL) THEN latestPupilCheck.mark
+      ELSE 0
+      END
+  ) AS DECIMAL(5,2)) / COUNT(p.id) AS [schoolAverage]
+FROM mtc_admin.pupil p
+-- FETCH COMPLETED PUPIL CHECK WITHIN THE CHECK WINDOW PERIOD
+LEFT OUTER JOIN
+(SELECT chk2.id, chk2.pupil_id, chk2.mark, ROW_NUMBER() OVER ( PARTITION BY chk2.pupil_id ORDER BY chk2.markedAt DESC ) as rank
+  FROM mtc_admin.[check] chk2
+  INNER JOIN mtc_admin.checkStatus cs ON cs.id = chk2.checkStatus_id
+  INNER JOIN mtc_admin.checkWindow cw ON cw.id = chk2.checkWindow_id
+  WHERE cs.code = 'CMP'
+        AND GETUTCDATE() BETWEEN cw.checkStartDate AND cw.adminEndDate
```

Comment:
what if i run this query in november, and i want to see the results from April earlier in the year?

Reply:
OK. but if you need to refresh the figures at any point outside of the check window, this query won't work.  I would modify the query to remove the hardcoded check window and use a `WHERE` constraint when you call it to filter to the check window you are targeting.  That way the query can be used at any time.

Reply:
@GuyHarwood resolving as there is a newer query for review